### DISCUSSION TITLE
Sorting similar results after filtering

### DISCUSSION THREAD

**Discussion by staticvoidzach** [johnlindquist/kit#1302] [Q&A]
I'm asking for a choice with `arg()` and have some options that are very similar, e.g.:
```
> Fire
Fireworks
Firefighters
```

However, I want to have "Firefighters" display first, as 9 times out of ten I will want "Firefighters" instead of "Fireworks".

Is there a way to supply something along the lines of a "weight" or "sortOrder" value or a custom ordering function so that after the relevance of my search has been applied, similar results are ordered how I would like?

**Comment by johnlindquist** [comment]
@staticvoidzach With the latest builds, you can use `groupChoices`:

```js
let choices = [
  {
    group: "Favorite",
    name: "Firefighters",
  },
  {
    group: "None",
    name: "Fireworks",
  },
]

let groupedChoices = groupChoices(choices)

let result = await arg("Select", groupedChoices)
```

`Favorite` will be pulled to the top by default. You can also provide an `order`:

```js

let choices = [
  {
    group: "People",
    name: "Firefighters",
  },
  {
    group: "Products",
    name: "Fireworks",
  },
  {
    group: "Insect",
    name: "Fireflies",
  },
]

let groupedChoices = groupChoices(choices, {
  order: ["People", "Insect", "Products"],
})

let result = await arg("Select", groupedChoices)
```
![CleanShot 2023-07-11 at 08 08 03](https://github.com/johnlindquist/kit/assets/36073/bfd40dd4-f545-4495-a561-c33bbdab4707)
